huaweicloud/services/iotda/resource_huaweicloud_iotda_dataforwarding_rule.go

```diff
@@ -0,0 +1,732 @@
+								Schema: map[string]*schema.Schema{
+									"region": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"project_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"bucket": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"custom_directory": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Computed: true,
+										ValidateFunc: validation.All(
+											validation.StringLenBetween(0, 256),
+											validation.StringMatch(regexp.MustCompile(`^[A-Za-z-_0-9/{}]*$`),
+												"Only letters, digits, hyphens (-), underscores (_), slash (/)"+
+													" and braces ({}) are allowed"),
+											validation.StringDoesNotMatch(regexp.MustCompile(`^/|/$|//`),
+												"Can not start or end with slashes (/), and cannot contain more than"+
+													" two adjacent slashes (/)"),
+										),
+									},
+								},
+							},
+						},
+
+						"amqp_forwarding": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"queue_name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+
+						"kafka_forwarding": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"region": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"project_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"addresses": {
+										Type:     schema.TypeList,
+										Required: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"ip": {
+													Type:     schema.TypeString,
+													Optional: true,
+												},
+
+												"domain": {
+													Type:     schema.TypeString,
+													Optional: true,
+												},
+
+												"port": {
+													Type:     schema.TypeInt,
+													Required: true,
+												},
+											},
+										},
+									},
+
+									"topic": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"user_name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"password": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										Sensitive: true,
+									},
+								},
+							},
+						},
+
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"space_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
```

Comment:
move this above the blocks to make the code easy to read

Reply:
ok